### PR TITLE
Create ModulePlan and Refactor

### PR DESCRIPTION
**Pull request by xCOLOURx**
#106 

Tests for EditCommand and FindCommand still require refactoring to ModulePlan.
Storage for ModulePlan not yet implemented.

### CODE REVIEW

File: src/test/java/seedu/address/model/ModulePlanTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        assertTrue(modulePlan.hasModule(CS2040S));
+    }
+
+    @Test
+    public void hasModule_moduleWithSameIdentityFieldsInModulePlan_returnsTrue() {
+        modulePlan.addModule(CS2040S);
+        Module editedCS2040S = new ModuleBuilder()
+                .withCode("CS2040S")
+                .withYear("1")
+                .withSem("1")
```
Comment (yyyaohhh) [inline src/test/java/seedu/address/model/ModulePlanTest.java:88]: The semester input here does not correspond to the format after changing

File: src/main/java/seedu/address/model/module/UniqueModuleList.java [hunk 2]
```diff
@@ -105,7 +105,7 @@ public Module find(ModuleCode code) {
                 return mods[i];
             }
         }
-        throw new ModuleNotFoundException();
+        return null;
```
Comment (yyyaohhh) [inline src/main/java/seedu/address/model/module/UniqueModuleList.java:108]: JavaDocs at line 98 can be removed accordingly

File: src/test/java/seedu/address/model/ModulePlanTest.java (hunk 1 above)
Comment (xCOLOURx) [reply src/test/java/seedu/address/model/ModulePlanTest.java]: ok fixed

File: src/main/java/seedu/address/model/module/UniqueModuleList.java (hunk 2 above)
Comment (xCOLOURx) [reply src/main/java/seedu/address/model/module/UniqueModuleList.java:107]: Ok removed.

**Comment by yyyaohhh** [comment]
LGTM👍

**Pull request merged**